src/main.js

```diff
@@ -336,19 +336,35 @@ class MiniGraphCard extends LitElement {
       </div>` : '';
   }
 
+  generateLegend(entity, index) {
+    let legend = this.computeName(index);
+    const state = this.getEntityState(index);
+    const { show_legend_state } = this.config.entities[index];
+
+    if (show_legend_state) {
+      legend += ` <span class="state__value">(${this.computeState(state)}</span><span class="state__uom">${this.computeUom(index)})})</span>`;
+    }
+
+    return legend;
+  }
+
   renderLegend() {
     if (this.visibleLegends.length <= 1 || !this.config.show.legend) return;
+
     return html`
       <div class="graph__legend">
-        ${this.visibleLegends.map(entity => html`
-          <div class="graph__legend__item"
-            @click=${e => this.handlePopup(e, this.entity[entity.index])}
-            @mouseenter=${() => this.setTooltip(entity.index, -1, this.getEntityState(entity.index), 'Current')}
-            @mouseleave=${() => (this.tooltip = {})}>
-            ${this.renderIndicator(this.getEntityState(entity.index), entity.index)}
-            <span class="ellipsis">${this.computeName(entity.index)}</span>
-          </div>
-        `)}
+        ${this.visibleLegends.map((entity) => {
+    const legend = this.generateLegend(entity, entity.index);
```

Comment:
See a remark about the `entity` variable above.

Reply:
The remark ?

Reply:
It was deleted after commits.
Remark is that I see no need of passing `entities` variable into `generateLegend()`

Reply:
Indeed fixed